ADAL/src/workplacejoin/ADPkeyAuthHelper.m

```diff
@@ -71,8 +71,7 @@ + (nullable NSString*)createDeviceAuthResponse:(nonnull NSString*)authorizationS
 {
     ADAuthenticationError* adError = nil;
     ADRegistrationInformation *info =
-    [ADWorkPlaceJoinUtil getRegistrationInformation:context
-                                              error:&adError];
+    [ADWorkPlaceJoinUtil getRegistrationInformation:context urlChallenge:nil error:&adError];
     
     if (!info && adError)
```

Comment:
Q: are we doing more harm by doing adError check here?
For whatever reason, what if info = nil and error is also nil?

Reply:
The reason for the current if condition is because we want to stop execution only in the case where info is nil and there is a keychain error incase of a corrupted certificate which will set adError.

In all other cases including the one where info and error are both nil, we still want to continue in our code path to create a response with blank pKeyAuthHeader and allowing the developers to enroll their device.